### PR TITLE
test: add vitest benchmarks

### PR DESCRIPTION
**Pull request by alephpiece**
### What this PR does

Adds vitest bench support for benchmarking performance-critical code paths.

For example:

```
 ✓  renderer  src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts > computeDisplayMessages Performance > 100 messages 1514ms     
     name                                       hz     min     max    mean     p75     p99    p995    p999     rme  samples
   · spread + reverse (O(n) copy)       765,103.85  0.0012  1.4548  0.0013  0.0013  0.0017  0.0018  0.0052  ±0.71%   382552
   · in-place backward index (no copy)  903,032.19  0.0010  0.4413  0.0011  0.0011  0.0014  0.0016  0.0039  ±0.47%   451517

 ✓  renderer  src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts > computeDisplayMessages Performance > 1000 messages 1442ms    
     name                                       hz     min     max    mean     p75     p99    p995    p999     rme  samples
   · spread + reverse (O(n) copy)       370,730.67  0.0023  0.5539  0.0027  0.0026  0.0035  0.0046  0.0259  ±0.74%   185366
   · in-place backward index (no copy)  904,676.55  0.0010  2.6768  0.0011  0.0011  0.0014  0.0016  0.0056  ±1.16%   452339

 ✓  renderer  src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts > computeDisplayMessages Performance > 10000 messages 1394ms   
     name                                       hz     min     max    mean     p75     p99    p995    p999     rme  samples
   · spread + reverse (O(n) copy)        57,508.57  0.0138  5.4271  0.0174  0.0152  0.0736  0.0851  0.2539  ±2.37%    28755
   · in-place backward index (no copy)  877,488.42  0.0010  0.4441  0.0011  0.0011  0.0015  0.0017  0.0104  ±0.57%   438745

 BENCH  Summary

   renderer  in-place backward index (no copy) - src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts > computeDisplayMessages Performance > 100 messages
    1.18x faster than spread + reverse (O(n) copy)

   renderer  in-place backward index (no copy) - src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts > computeDisplayMessages Performance > 1000 messages
    2.44x faster than spread + reverse (O(n) copy)

   renderer  in-place backward index (no copy) - src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts > computeDisplayMessages Performance > 10000 messages
    15.26x faster than spread + reverse (O(n) copy)
```

Fixes #

### Why we need it and why it was done in this way

The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Breaking changes

If this PR introduces breaking changes, please describe the changes and the impact on users.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Documentation: A [user-guide update](https://docs.cherry-ai.com) was considered and is present (link) or not required. Check this only when the PR introduces or changes a user-facing feature or behavior.
- [ ] Self-review: I have reviewed my own code (e.g., via [`/gh-pr-review`](/.claude/skills/gh-pr-review/SKILL.md), `gh pr diff`, or GitHub UI) before requesting review from others

### Release note

```release-note
Add vitest bench support for benchmarking performance-critical code paths.
```

### CODE REVIEW

File: src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+const mockDataMap = Object.fromEntries(SCENARIOS.map((n) => [n, generateMockMessages(n)])) as Record<
+  (typeof SCENARIOS)[number],
+  Message[]
+>
+
+// 测试结果是否一致
+test('computeOld and computeNew should produce identical results', () => {
+  const sample = mockDataMap[100]
+  expect(baseline(sample, 0, 20)).toEqual(byBackwardIndex(sample, 0, 20))
+})
```
Comment (GeorgeDong32) [inline src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts:112]: **Suggestion: Add more edge case tests**

The current test only validates one scenario (`100 messages, startIndex=0, displayCount=20`). Consider adding tests for:

```typescript
describe('algorithm correctness', () => {
  test('empty messages', () => {
    expect(baseline([], 0, 20)).toEqual(byBackwardIndex([], 0, 20))
  })
  
  test('startIndex at boundary', () => {
    const sample = mockDataMap[100]
    expect(baseline(sample, 50, 20)).toEqual(byBackwardIndex(sample, 50, 20))
    expect(baseline(sample, 99, 20)).toEqual(byBackwardIndex(sample, 99, 20))
  })
  
  test('displayCount larger than remaining', () => {
    const sample = mockDataMap[100]
    expect(baseline(sample, 0, 200)).toEqual(byBackwardIndex(sample, 0, 200))
  })
})
```

This ensures correctness across all code paths, not just the happy path.

File: src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      return
+    }
+    displayMessages.push(message)
+  }
+
+  for (let i = startIndex; i < reversedMessages.length && userIdSet.size + assistantIdSet.size < displayCount; i++) {
+    processMessage(reversedMessages[i])
+  }
+
+  return displayMessages
```
Comment (GeorgeDong32) [inline src/renderer/src/pages/home/Messages/__tests__/Messages.bench.ts:37]: **Note: Code duplication for testing purposes**

The `processMessage` function is duplicated between `baseline` and `byBackwardIndex` (lines 20-31 and 54-65). This is intentional for benchmark isolation - each implementation must be independent to measure performance accurately.

However, consider extracting the shared logic for the correctness test, or add a comment explaining why duplication is intentional:

```typescript
// Note: processMessage is intentionally duplicated in each implementation
// to ensure fair benchmark comparison without shared function call overhead
```

File: src/renderer/src/pages/home/Messages/Messages.tsx [hunk 3]
```diff
@@ -349,13 +349,14 @@ const Messages: React.FC<MessagesProps> = ({ assistant, topic, setActiveTopic, o
...
-  // 如果剩余消息数量小于 displayCount，直接返回所有剩余消息
-  if (reversedMessages.length - startIndex <= displayCount) {
-    return reversedMessages.slice(startIndex)
+  // 如果剩余消息数量小于 displayCount，直接返回所有剩余消息的倒序切片
+  if (messages.length - startIndex <= displayCount) {
+    const result: Message[] = []
+    for (let i = messages.length - 1 - startIndex; i >= 0; i--) {
+      result.push(messages[i])
+    }
+    return result
```
Comment (GeorgeDong32) [inline src/renderer/src/pages/home/Messages/Messages.tsx:358]: **✅ Good performance optimization**

The refactored early-return branch avoids the O(n) spread + reverse copy by using in-place backward iteration. This is a clean, well-documented optimization.

Key improvements:
- Eliminates `[...messages].reverse()` which creates a full copy
- Uses direct index calculation: `messages.length - 1 - startIndex`
- Clear Chinese comments explaining the optimization

The benchmark results in the PR description (up to 15x faster for 10k messages) validate the effectiveness of this change.

**Review by DeJeune — APPROVED** [review]
LGTM — clean performance optimization. The backward index traversal correctly mirrors the old reverse+slice semantics and the benchmark validates equivalence.
